src/applications/disability-benefits/2346/Components/ConfirmAddress.jsx

```diff
@@ -0,0 +1,73 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { getVeteranAddressInfo } from '../actions';
+
+class ConfirmAddress extends Component {
+  static propTypes = {
+    addressLine1: PropTypes.string.isRequired,
+    addressLine2: PropTypes.string.isRequired,
+    city: PropTypes.string.isRequired,
+    state: PropTypes.string.isRequired,
+    country: PropTypes.string.isRequired,
+    zip: PropTypes.string.isRequired,
+    email: PropTypes.string.isRequired,
+    getVeteranAddressInfo: PropTypes.func.isRequired,
+  };
+  static defaultProps = {
+    addressLine1: '',
+    addressLine2: '',
+    city: '',
+    state: '',
+    country: '',
+    zip: '',
+    email: '',
+    getVeteranAddressInfo: () => {},
+  };
+  componentDidMount() {
+    this.props.getVeteranAddressInfo();
+  }
+
+  render() {
+    const { addressLine1, addressLine2, city, state, zip, email } = this.props;
+
+    return (
+      <div>
+        <h1>Confirm your address</h1>
+        <div>
+          <label htmlFor="address1">Address Line 1</label>
+          <input id="address1" type="text" defaultValue={addressLine1} />
+          <label htmlFor="address2">Address Line 2</label>
+          <input type="text" defaultValue={addressLine2} />
+          <label htmlFor="city">City</label>
+          <input type="text" id="city" defaultValue={city} />
+          <label htmlFor="state">State</label>
+          <input type="text" id="state" defaultValue={state} />
+          <label htmlFor="zip">Zip</label>
+          <input type="text" id="zip" defaultValue={zip} />
+          <p className="vads-u-margin-top--4">Confirm your email address</p>
+          <input type="text" defaultValue={email} />
+        </div>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  addressLine1: state.form2346Reducer?.formData?.veteranAddress?.street,
+  addressLine2: state.form2346Reducer?.formData?.veteranAddress?.street,
+  city: state.form2346Reducer?.formData?.veteranAddress?.city,
+  state: state.form2346Reducer?.formData?.veteranAddress?.state,
+  country: state.form2346Reducer?.formData?.veteranAddress?.country,
+  zip: state.form2346Reducer?.formData?.veteranAddress?.postalCode,
+  email: state.form2346Reducer?.formData?.email,
+});
+
+const mapDispatchToProps = {
+  getVeteranAddressInfo,
+};
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps,
+)(ConfirmAddress);
```

Comment:
I'm not sure exactly how this Address component is supposed to function as far as backend, but if you're interested in updating this data in the user's profile (in a data service known as VA Profile, formerly Vet360), you should explore these [React components](https://github.com/department-of-veterans-affairs/vets-website/tree/master/src/platform/user/profile/vet360). Those profile updates are actually very complex, but should be trivial using these components.

Reply:
This is purely to prefill the veteran's address data within the form fields. . But thanks @ncksllvn glad to be made aware of this because in our next couple of sprint giving the veteran the ability to update a forwarding address to ship to is some functionality we have to implement.

Reply:
Hey @ncksllvn Shouldn't there be some functionality across vets-website to have the option to have prefill form functionality for veteran profile data?

Reply:
Echoing what Nick said. I also don't have the full picture re: what you're doing here, but it might be worth looking at how the Letters app is using the MailingAddress component to both show the mailing address that's on file and also allow the Vet to update their address

https://github.com/department-of-veterans-affairs/vets-website/blob/cfb7e00886cad4297514fbbc06312ab6d459c89b/src/applications/letters/containers/AddressSection.jsx#L29-L47

Reply:
@erikphansen What form is this use on?